File: packages/api/scripts/smoke.py
"""端到端煙霧測試：不開 HTTP，直接跑多輪對話。

驗證 slot filling -> 媒合代理 -> 報價 -> 建立預約單 -> 偏好記錄 都真的會發生。

執行（從 repo 根目錄）：
    .venv\\Scripts\\python.exe packages\\api\\scripts\\smoke.py
"""

from __future__ import annotations

import sys
import time
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parents[1]))

from op_agent.domain import PERIOD_LABEL  # noqa: E402
from op_agent.seed import DEMO_USER_ID  # noqa: E402
from op_agent.user_agent import run_user_agent_turn  # noqa: E402

TURNS = [
    "我家冷氣不冷了，主臥那台",
    "台北市大安區復興南路一段100號5樓，下午方便",
    "預算不要太高，幫我看有哪些選擇",
    "那就約冷研家電",
]


def main() -> int:
    session_id: str | None = None

    for i, message in enumerate(TURNS, start=1):
        print("\n" + "=" * 72)
        print(f"[第 {i} 輪] 會員：{message}")
        print("=" * 72)

        started = time.monotonic()
        try:
            r = run_user_agent_turn(
                session_id=session_id, inbr_account_id=DEMO_USER_ID, message=message
            )
        except Exception as err:  # noqa: BLE001
            print(f"  !! 失敗：{type(err).__name__}: {err}")
            return 1

        session_id = r["sessionId"]

        for t in r.get("trace", []):
            print(f"  · [{t['agent']}] {t['tool']}")

        print(f"\n管家：{r['reply']}\n")

        req = r.get("request")
        if req:
            addr = req.get("address")
            print(
                f"  服務單 {req['requestId']}  狀態={req.get('status')}"
                f"  症狀={'/'.join(req.get('slots', {}).get('symptoms') or []) or '-'}"
                f"  地址={(addr.get('countyName', '') + addr.get('districtName', '')) if addr else '-'}"
                f"  時段={PERIOD_LABEL.get(req.get('preferredContactTime', ''), '-')}"
            )

        match = r.get("match")
        if match:
            print(f"  媒合結果（{len(match.get('proposals', []))} 家）：")
            for p in match.get("proposals", []):
                q = p["quote"]
                print(
                    f"    - {p['vendorName']}  分數 {p['score']}"
                    f"  {q['estimatedMin']}~{q['estimatedMax']} 元"
                    f"  最快 {p['earliestSlot']['date']}"
                )
            print(f"  媒合總結：{match.get('summary')}")

        booking = r.get("booking")
        if booking:
            print(
                f"  預約單 {booking['orderNo']}  {booking['vendorName']}"
                f"  {booking['serviceDate']} {PERIOD_LABEL.get(booking['servicePeriod'], '')}"
                f"  訂金 {booking['depositAmount']} 元  狀態 {booking['orderStatus']}"
            )

        print(f"  偏好：{r.get('preferences')}")
        print(f"  ({int((time.monotonic() - started) * 1000)}ms)")

    print("\n煙霧測試結束")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: packages/backend/src/scripts/smoke.ts
/**
 * 端到端煙霧測試：不開 HTTP，直接跑多輪對話，驗證
 *   slot filling -> 媒合代理 -> 報價 -> 建立預約單 -> 偏好記錄
 * 都真的會發生。
 *
 * 執行：npm run smoke -w @op/backend
 */
import { readFileSync } from 'node:fs';
import { resolve } from 'node:path';

for (const p of ['.env', '../../.env']) {
  try {
    const text = readFileSync(resolve(process.cwd(), p), 'utf8');
    for (const line of text.split(/\r?\n/)) {
      const m = /^\s*([A-Z0-9_]+)\s*=\s*(.*)\s*$/.exec(line);
      if (m && process.env[m[1]] === undefined) process.env[m[1]] = m[2].replace(/^['"]|['"]$/g, '');
    }
    break;
  } catch {
    /* try next */
  }
}

const { runUserAgentTurn } = await import('../agents/userAgent');
const { DEMO_USER_ID } = await import('../data/users');

const TURNS = [
  '我家冷氣不冷了，主臥那台',
  '台北市大安區復興南路一段100號5樓，下午方便',
  '我想選最便宜的那家',
];

let sessionId: string | undefined;

for (const [i, message] of TURNS.entries()) {
  console.log(`\n${'='.repeat(70)}\n[第 ${i + 1} 輪] 會員：${message}\n${'='.repeat(70)}`);
  const started = Date.now();
  const r = await runUserAgentTurn({ sessionId, inbrAccountId: DEMO_USER_ID, message });
  sessionId = r.sessionId;

  for (const t of r.trace) {
    console.log(`  · [${t.agent}] ${t.tool}`);
  }
  console.log(`\n管家：${r.reply}\n`);
  console.log(
    `  服務單狀態：${r.request?.status ?? '-'}` +
      `  症狀=${r.request?.slots.symptoms?.join('/') ?? '-'}` +
      `  地址=${r.request?.address ? r.request.address.countyName + r.request.address.districtName : '-'}` +
      `  時段=${r.request?.preferredContactTime ?? '-'}`,
  );
  if (r.match) {
    console.log(`  媒合結果（${r.match.proposals.length} 家）：`);
    for (const p of r.match.proposals) {
      console.log(
        `    - ${p.vendorName}  分數 ${p.score}  ${p.quote.estimatedMin}~${p.quote.estimatedMax} 元  最快 ${p.earliestSlot.date}`,
      );
    }
  }
  if (r.booking) {
    console.log(
      `  預約單：${r.booking.orderNo}  ${r.booking.vendorName}  ${r.booking.serviceDate}  訂金 ${r.booking.depositAmount} 元  狀態 ${r.booking.orderStatus}`,
    );
  }
  console.log(`  (${Date.now() - started}ms)`);
}

console.log('\n煙霧測試結束');
